proposals/pattern-variables.md

```diff
@@ -0,0 +1,73 @@
+# Variable declarations under disjunctive patterns
+
+## Summary  
+
+Allow variable declarations under `or` patterns and across `case` labels in a `switch` section.
+
+## Motivation
```

Comment:
I'd add some more practical (less abstract) motivating examples. 
Looking that the roslyn codebase, we often have a `switch` expression just to get some value out of different kinds of bound nodes. Many can be found just by searching for ["node switch"](https://grep.app/search?q=%20node%20switch&case=true&filter[repo][0]=dotnet/roslyn).
But I have to say looking at that search result, most of them don't look like they would benefit from this feature (either they use `when` clauses for different cases, or they call different property/method overloads in the different cases).